src/test/java/com/easypost/BatchTest.java

```diff
@@ -0,0 +1,199 @@
+     */
+    @Test
+    public void testAll() throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+        
+        params.put("page_size", Fixture.pageSize());
+
+        BatchCollection batches = Batch.all(params);
+
+        assertNotNull(batches.getHasMore());
+        assertTrue(batches.getBatches().size() <= 5);
+        for(Batch batch: batches.getBatches()) {
+            assertTrue(batch.getId().startsWith("batch_"));
+        }
+    }
+
+    /**
+     * Test creating and buying a Batch in a single call.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testCreateAndBuy() throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+        List<Object> shipmentData = new ArrayList<>();
+
+        shipmentData.add(Fixture.oneCallBuyShipment());
+        params.put("shipments", shipmentData);
+
+        Batch batch = Batch.createAndBuy(params);
+
+        assertTrue(batch instanceof Batch);
+        assertTrue(batch.getId().startsWith("batch_"));
+        assertEquals(1, batch.getNumShipments().intValue());
+    }
+
+    /**
+     * Test buying a batch.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    @Disabled
+    public void testBuy() throws EasyPostException {
+        List<Object> shipmentData = new ArrayList<>();
+        Map<String, Object> shipments = new HashMap<>();
+
+        shipmentData.add(Fixture.oneCallBuyShipment());
+        shipments.put("shipments", shipmentData);
+
+        Batch batch = Batch.create(shipments);
+
+        batch = batch.buy();
+
+        assertTrue(batch instanceof Batch);
+        assertTrue(batch.getId().startsWith("batch_"));
+        assertEquals(1, batch.getNumShipments().intValue());
+    }
+
+    /**
+     * Test creating a scanform for a batch.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    public void testCreateScanForm() throws EasyPostException {
+        Batch batchWithScanForm = batch.createScanForm();
+
+        // We can't assert anything meaningful here because the scanform gets queued for generation and may not be immediately available
+        assertTrue(batchWithScanForm instanceof Batch);
+    }
+
+    /**
+     * Test adding and removing a shipment from a batch.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testAddRemoveShipment() throws EasyPostException {
+        Shipment shipment = Shipment.create(Fixture.oneCallBuyShipment());
+        
+        Batch batch = Batch.create();
+
+        List<Shipment> shipmentData = new ArrayList<>();
+        Map<String, Object> params = new HashMap<>();
+
+        shipmentData.add(shipment);
+        params.put("shipments", shipmentData);
+
+        Batch batchWithAddedShipment = batch.addShipments(params);
+
+        assertTrue(batchWithAddedShipment.getId().startsWith("batch_"));
```

Comment:
Please follow the PHP library's test suite. Asserting the ID here isn't necessary as we aren't doing anything related to the ID - we are only adding and removing something from the object and these assertions are handled elsewhere in other tests.